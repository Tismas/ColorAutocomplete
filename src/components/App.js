import React, { Component } from 'react';
import { connect } from 'react-redux';

import { fetchColors } from '../store/actions/autocompleteActions';
import { changeInput, showPreview, hidePreview, submitColor } from '../store/actions/colorsActions'

import AutoSuggest from './AutoSuggest';

@connect(store => ({
    suggestions: store.autocomplete.colors,
    autocompleteFetched: store.autocomplete.fetched,
    autocompleteFetching: store.autocomplete.fetching,
    autocompleteFetchError: store.autocomplete.error,
    ...store.colors
}))
class App extends Component {
    constructor() {
        super();
        this.onColorInputChange = this.onColorInputChange.bind(this);
        this.showPreview = this.showPreview.bind(this);
        this.hidePreview = this.hidePreview.bind(this);
        this.submitColor = this.submitColor.bind(this);
    }
    componentWillMount() {
        this.props.dispatch(fetchColors());
    }

    onColorInputChange(value) {
        this.props.dispatch(changeInput( value ));
    }

    showPreview(overlayColor) {
        this.props.dispatch(showPreview( overlayColor ));
    }

    hidePreview() {
        this.props.dispatch(hidePreview());
    }

    submitColor() {
        this.props.dispatch(submitColor( this.props.overlayColor ));
    }

    render() {
        return (
            <div className="bg">
                <div
                    className="bg-overlay"
                    style={{
                        '--background-overlay-color': this.props.overlayColor,
                        opacity: this.props.colorPreview ? 0.5 : 0.8
                    }}
                /> 
                <div className="input-container">
                    <AutoSuggest
                        value={this.props.searchValue}
                        suggestions={this.props.suggestions}

                        onChange={this.onColorInputChange}
                        onSuggestionHover={this.showPreview}
                        onSuggestionLeaveHover={this.hidePreview}
                    />
                    <button className="accept-button" onClick={this.submitColor}> Accept </button>
                </div>
            </div>
        );
    }
}

export default App;
