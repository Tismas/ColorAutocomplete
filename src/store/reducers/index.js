import { combineReducers } from 'redux';

import colorAutocompleteReducer from './colorAutocompleteReducer';
import colorsReducer from './colorsReducer';

export default combineReducers({
    autocomplete: colorAutocompleteReducer,
    colors: colorsReducer
});
