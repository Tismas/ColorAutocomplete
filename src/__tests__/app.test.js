import React from 'react';
import renderer from 'react-test-renderer';

import App from '../components/App';
import { Provider } from 'react-redux';
import store from '../store';

test('Render of the whole app', () => {
  const component = renderer.create(
    <Provider store={store}>
        <App />
    </Provider>, 
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});