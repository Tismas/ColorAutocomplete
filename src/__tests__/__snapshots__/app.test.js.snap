// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render of the whole app 1`] = `
<div
  className="bg"
>
  <div
    className="bg-overlay"
    style={
      Object {
        "--background-overlay-color": "#fff",
        "opacity": 0.8,
      }
    }
  />
  <div
    className="input-container"
  >
    <div
      className="autosuggest-wrapper"
    >
      <input
        className="autosuggest-input "
        onChange={[Function]}
        value=""
      />
    </div>
    <button
      className="accept-button"
      onClick={[Function]}
    >
       Accept 
    </button>
  </div>
</div>
`;
